AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/QtPyScriptCanvasNodePalette.py

```diff
@@ -0,0 +1,47 @@
+"""
+Copyright (c) Contributors to the Open 3D Engine Project.
+For complete copyright and license terms please see the LICENSE at the root of this distribution.
+
+SPDX-License-Identifier: Apache-2.0 OR MIT
+
+Object to house all the Qt Objects and behavior used in testing the script canvas node palette
+"""
+from editor_python_test_tools.utils import TestHelper as helper
+from PySide2 import QtWidgets, QtTest, QtCore
+import pyside_utils
+from consts.scripting import (NODE_PALETTE_QT, TREE_VIEW_QT)
+from consts.general import (WAIT_TIME_SEC_3)
+
+class QtPyScriptCanvasNodePalette():
+    """
+       QtPy class for handling the behavior of the script canvas node palette
+        """
+    def __init__(self, sc_editor):
+        self.node_palette = sc_editor.sc_editor.findChild(QtWidgets.QDockWidget, NODE_PALETTE_QT)
+        self.node_palette_tree = self.node_palette.findChild(QtWidgets.QTreeView, TREE_VIEW_QT)
+        self.node_palette_search_frame = self.node_palette.findChild(QtWidgets.QFrame, "searchFrame")
```

Comment:
`searchFrame` should be a const somewhere.

Reply:
this actually was a constant but I forgot to bring it in.